app/scripts/lib/setupWeb3.js

```diff
@@ -31,8 +36,117 @@ export default function setupWeb3(log) {
     value: web3.eth,
   })
 
+  // Setup logging of nested property usage
+  if (shouldLogUsage) {
+    // web3 namespaces with common and uncommon dapp actions
+    const includedTopKeys = [
+      'eth',
+      'db',
+      'shh',
+      'net',
+      'personal',
+      'bzz',
+      'version',
+    ]
+
+    // For each top-level property, create appropriate Proxy traps for all of
+    // their properties
+    includedTopKeys.forEach((topKey) => {
+      const applyTrapKeys = new Map()
+      const getTrapKeys = new Map()
+
+      Object.keys(web3[topKey]).forEach((key) => {
+        const path = `web3.${topKey}.${key}`
+
+        if (web3Entitites[path]) {
+          if (web3Entitites[path] === 'function') {
+            applyTrapKeys.set(key, path)
+          } else {
+            getTrapKeys.set(key, path)
+          }
+        }
+      })
+
+      // Create apply traps for function properties
+      for (const [key, path] of applyTrapKeys) {
+        web3[topKey][key] = new Proxy(web3[topKey][key], {
+          apply: (...params) => {
+            window.ethereum.request({
+              method: 'metamask_logInjectedWeb3Usage',
+              params: [
+                {
+                  action: 'apply',
+                  path,
+                },
+              ],
+            })
+
+            // Call function normally
+            return Reflect.apply(...params)
+          },
+        })
+      }
+
+      // Create get trap for non-function properties
+      web3[topKey] = new Proxy(web3[topKey], {
+        get: (web3Prop, key, ...params) => {
+          const name = stringifyKey(key)
+
+          if (getTrapKeys.has(name)) {
+            window.ethereum.request({
+              method: 'metamask_logInjectedWeb3Usage',
+              params: [
+                {
+                  action: 'get',
+                  path: getTrapKeys.get(name),
+                },
+              ],
+            })
+          }
+
+          // return value normally
+          return Reflect.get(web3Prop, key, ...params)
+        },
+      })
+    })
+  }
+
+  const unEnumerableFunctionKeys = [
+    'isConnected',
+    'setProvider',
+    'reset',
+    'toHex',
+    'fromAscii',
+    'fromDecimal',
+    'toWei',
+    'isAddress',
```

Comment:
I found a few keys in the web3 docs that weren't included here:

```suggestion
    'sha3',
    'toHex',
    'toAscii',
    'fromAscii',
    'toDecimal',
    'fromDecimal',
    'fromWei',
    'toWei',
    'toBigNumber',
    'isAddress',
```